tests/plugins/test_add_labels.py

```diff
@@ -354,3 +358,48 @@ def test_dont_overwrite_distribution_scope(tmpdir, docker_tasker, parent_scope,
 
     result = df.labels.get("distribution-scope")
     assert result == result_scope
+
+
+@pytest.mark.parametrize('url_format, info_url, expected_log', [
+    ('url_pre {label1} {label2} url_post', 'url_pre label1_value label2_value url_post', None),
+    ('url_pre url_post', 'url_pre url_post', None),
+    ('url_pre {label1} {label2} {label3_non_existent} url_post', None, "KeyError:"),
+])
+def test_url_label(tmpdir, docker_tasker, caplog, url_format, info_url, expected_log):
```

Comment:
Checking log is not sufficient, we should be checking that exception is not just logged, but actually thrown